src/tests/system.rs

```diff
@@ -607,3 +607,148 @@ mod fixed {
+                    == Length::new::<meter>(*v).is_normal()
+            }
+
+            #[cfg(feature = "std")]
+            #[allow(trivial_casts)]
+            fn cbrt(v: A<V>) -> bool {
+                let l: Quantity<Q<P1, Z0, Z0>, U<V>, V> = Quantity::<Q<P3, Z0, Z0>, U<V>, V> {
+                    dimension: PhantomData,
+                    units: PhantomData,
+                    value: *v,
+                }.cbrt();
+
+                Test::eq(&v.cbrt(), &l.value)
+            }
+
+            #[cfg(feature = "std")]
+            #[allow(trivial_casts)]
+            fn hypot(l: A<V>, r: A<V>) -> bool {
+                Test::eq(&Length::new::<meter>(l.hypot(*r)),
+                    &Length::new::<meter>(*l).hypot(Length::new::<meter>(*r)))
+            }
+
+            #[cfg(feature = "std")]
+            #[allow(trivial_casts)]
+            fn mul_add(s: A<V>, a: A<V>, b: A<V>) -> bool {
+                let r: Quantity<Q<P2, Z0, Z0>, U<V>, V> = Length::new::<meter>(*s).mul_add(
+                    Length::new::<meter>(*a),
+                    Quantity::<Q<P2, Z0, Z0>, U<V>, V> {
+                        dimension: PhantomData,
+                        units: PhantomData,
+                        value: *b
+                    });
+
+                Test::eq(&s.mul_add(*a, *b), &r.value)
+            }
+
+            #[allow(trivial_casts)]
+            fn recip(v: A<V>) -> bool {
+                let a: Quantity<Q<N1, Z0, Z0>, U<V>, V> = Quantity::<Q<P1, Z0, Z0>, U<V>, V> {
+                    dimension: PhantomData,
+                    units: PhantomData,
+                    value: *v,
+                }.recip();
+
+                Test::eq(&v.recip(), &a.value)
+            }
+
+            #[cfg(feature = "std")]
+            #[allow(trivial_casts)]
+            fn powi(v: A<V>) -> bool {
+                Test::eq(&v.powi(3), &Length::new::<meter>(*v).powi(P3::new()).value)
+            }
+
+            #[cfg(feature = "std")]
+            #[allow(trivial_casts)]
+            fn sqrt(v: A<V>) -> TestResult {
+                if *v < V::zero() {
+                    return TestResult::discard();
+                }
+
+                let l: Quantity<Q<P1, Z0, Z0>, U<V>, V> = Quantity::<Q<P2, Z0, Z0>, U<V>, V> {
+                    dimension: PhantomData,
+                    units: PhantomData,
+                    value: *v,
+                }.sqrt();
+
+                TestResult::from_bool(Test::eq(&v.sqrt(), &l.value))
+            }
+
+            #[allow(trivial_casts)]
+            fn max(l: A<V>, r: A<V>) -> bool {
+                Test::eq(&Length::new::<meter>(l.max(*r)),
+                    &Length::new::<meter>(*l).max(Length::new::<meter>(*r)))
+            }
+
+            #[allow(trivial_casts)]
+            fn min(l: A<V>, r: A<V>) -> bool {
+                Test::eq(&Length::new::<meter>(l.min(*r)),
+                    &Length::new::<meter>(*l).min(Length::new::<meter>(*r)))
+            }
```

Comment:
Exactly duplicated from the `quickcheck!` block above for `Float`. Can this duplication be reduced?

Reply:
You can extract the duplicated code into a `storage_types!` block with `types: Complex, Float;`. For the `is_*` functions that are currently slightly different you can just use `Complex`'s `is_*` implementation. e.g. [`Complex::is_normal`](https://docs.rs/num-complex/0.4.0/src/num_complex/lib.rs.html#590) is already doing `v.re.is_normal() && v.im.is_normal()`

One thing to keep in mind when doing this is that `storage_types!` always generates modules with a name of the type. This means you can't execute `storage_types!` for the same types in the same parent module and may need to introduce a new child module.